src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/RazorCodeActionContext.cs

```diff
@@ -14,20 +15,21 @@ internal sealed class RazorCodeActionContext
             CodeActionParams request,
             DocumentSnapshot documentSnapshot,
             RazorCodeDocument codeDocument,
-            SourceLocation location,
+            SourceText sourceText,
             bool supportsFileCreation)
         {
             Request = request ?? throw new ArgumentNullException(nameof(request));
             DocumentSnapshot = documentSnapshot ?? throw new ArgumentNullException(nameof(documentSnapshot));
             CodeDocument = codeDocument ?? throw new ArgumentNullException(nameof(codeDocument));
-            Location = location;
+            SourceText = sourceText ?? throw new ArgumentNullException(nameof(sourceText));
             SupportsFileCreation = supportsFileCreation;
         }
 
         public CodeActionParams Request { get; }
         public DocumentSnapshot DocumentSnapshot { get; }
         public RazorCodeDocument CodeDocument { get; }
-        public SourceLocation Location { get; }
+        public SourceText SourceText { get; }
         public bool SupportsFileCreation { get; }
+        public SourceLocation Location { get; set; }
```

Comment:
Feels odd for this to be settable, left a comment above where it's used.

Reply:
Addressing comment above.